src/wrappers/themis/rust/src/keygen.rs

```diff
@@ -89,29 +89,29 @@ fn try_gen_rsa_key_pair() -> Result<RsaKeyPair> {
         }
     }
 
-    secret_key.reserve(secret_key_len);
-    public_key.reserve(secret_key_len);
+    private_key.reserve(private_key_len);
+    public_key.reserve(private_key_len);
```

Comment:
shouldn't it be `public_key_len`?

```suggestion
    public_key.reserve(public_key_len);
```

Reply:
Nice catch!

I love code review. Completely missed this sleeper bug. It seems the code and the debugging assertion that follows it work fine only because the private key is always longer than the public one.

There's a similar bug in ECDSA key generation. I'll address that as well.